src/Common/AsyncLoader.cpp

```diff
@@ -179,9 +168,9 @@ AsyncLoader::AsyncLoader(std::vector<PoolInitializer> pool_initializers, bool lo
             .thread_pool = std::make_unique<ThreadPool>(
                 init.metric_threads,
                 init.metric_active_threads,
-                init.max_threads,
-                /* max_free_threads = */ 0,
-                init.max_threads),
+                /* max_threads = */ std::numeric_limits<size_t>::max(), // Unlimited number of threads, we do worker management ourselves
```

Comment:
Should we pass some safety constraint just in case?

Reply:
I think it is not necessary. And it is hard to pick a value.

Reply:
> And it is hard to pick a value

Is it becasue of [this](https://github.com/ClickHouse/ClickHouse/pull/49351#discussion_r1317869814)?

Reply:
Yes, exactly due to possible suspended workers